rules/integrations/aws/aws_bedrock_high_resource_consumption_detection.toml

```diff
@@ -0,0 +1,54 @@
+[metadata]
+creation_date = "2024/05/04"
+maturity = "production"
+updated_date = "2024/05/04"
+min_stack_comments = "ES|QL rule type is still in technical preview as of 8.13, however this rule was tested successfully; integration in tech preview"
+min_stack_version = "8.13.0"
+
+[rule]
+author = ["Elastic"]
+description = """
+Detects potential resource exhaustion or data breach attempts by monitoring for users who consistently generate high input token counts, submit numerous requests, and receive large responses. This behavior could indicate an attempt to overload the system or extract an unusually large amount of data, possibly revealing sensitive information or causing service disruptions.
+"""
+false_positives = ["Authorized heavy usage of the system that is business justified and monitored."]
+from = "now-1d"
+interval = "10m"
+language = "esql"
+license = "Elastic License v2"
+name = "Potential Abuse of Resources by High Token Count and Large Response Sizes"
+references = [
+    "https://atlas.mitre.org/techniques/AML.T0051",
+    "https://owasp.org/www-project-top-10-for-large-language-model-applications/",
+    "https://www.elastic.co/security-labs/elastic-advances-llm-security",
+]
+risk_score = 47
+rule_id = "b1773d05-f349-45fb-9850-287b8f92f02d"
+setup = """## Setup
+
+This rule requires that guardrails are configured in AWS Bedrock. For more information, see the AWS Bedrock documentation:
+
+https://docs.aws.amazon.com/bedrock/latest/userguide/guardrails-create.html
+"""
+severity = "medium"
+tags = [
+    "Domain: LLM",
+    "Data Source: AWS Bedrock",
+    "Data Source: AWS S3",
+    "Use Case: Potential Overload",
+    "Use Case: Resource Exhaustion",
+    "Mitre Atlas: LLM04"
+]
+timestamp_override = "event.ingested"
+type = "esql"
+
+query = '''
```

Comment:
any suggested search time window ? (e.g if run for last 60m vs last 24h may impact the used thresholds) if so better hardcoded in the query (e.g. `| where  @timestamp > now() - ...`.)

Reply:
good question, I almost wonder what are the trade offs.

Reply:
I think we should avoid any hardcoded windowing since it can conflict with the rule timeing

lookback (from - to) / interval